### PR TITLE
Implemented new priority levels

### PR DESCRIPTION
**Pull request by jonasfj**
Please give this a detailed review... I was like 30% awake when I wrote 90% of this....

### CODE REVIEW

File: schemas/create-task-request.yml [hunk 1]
```diff
@@ -96,15 +96,21 @@ properties:
   priority:
     title:          "Task Priority"
     description: |
-      Priority of task, this defaults to `normal`. Additional levels may be
-      added later.
+      Priority of task, this defaults to `lowest`, equivalent to `normal` which
```
Comment (imbstack) [inline schemas/create-task-request.yml]: Perhaps "...equivalent to the deprecated `normal` level.

File: src/api.js [hunk 2]
```diff
@@ -572,29 +591,35 @@ api.declare({
...
+    `queue:create-task:${provisionerId}/${workerType}`,
+  ], [
+    `queue:define-task:${provisionerId}/${workerType}`,
+    `queue:task-group-id:${schedulerId}/${taskGroupId}`,
+    `queue:schedule-task:${schedulerId}/${taskGroupId}/${taskId}`,
+  ]], true);
+
+  // Require 'queue:create-task:<priority>:<provisionerId>/<workerType>' for
+  // given priority or higher...
+  let priorities = PRIORITY_LEVELS.slice(0, PRIORITY_LEVELS.indexOf(taskDef.priority) + 1);
```
Comment (imbstack) [inline src/api.js]: 😄

File: src/api.js [hunk 3]
```diff
@@ -752,28 +777,26 @@ api.declare({
...
+  assert(priorities.length > 0, 'must have a non-empty list of priorities');
+  // If not authorized by legacy, we require most recent scopes
+  if (!legacyAuthorized && !req.satisfies(priorities.map(priority => [
+    `queue:create-task:${priority}:${provisionerId}/${workerType}`,
+    `queue:scheduler-id:${schedulerId}`,
+  ]))) {
```
Comment (imbstack) [inline src/api.js]: We're missing the `taskDef.scopes` and `taskDef.routes` checks in `defineTask` now, right?

Comment (jonasfj) [inline src/api.js]: Good catch!

**Comment by jonasfj** [comment]
@imbstack, let roll this out Monday...

**Pull request merged**